src/util.ts

```diff
@@ -61,3 +64,39 @@ export function* zip<X, Y>(xs: Iterable<X>, ys: Iterable<Y>): Iterable<X | Y> {
     y1 = y.next()
   }
 }
+
+export async function persistentChatAction(
+  ctx: Context,
+  callback: () => Promise<void>,
+  action: ChatAction,
+  every: number = 8000
+) {
+  return new Promise<void>(async (resolve) => {
+    await ctx.sendChatAction(action)
+
+    const timer = setInterval(
+      async () => await ctx.sendChatAction(action),
+      every
+    )
+
+    await callback()
+    clearInterval(timer)
+    resolve()
+  })
+}
+
+export async function timedChatAction(ctx: any, action: ChatAction, duration: number, every?: number) {
```

Comment:
`timedChatAction` seems wasteful; not clear where it can be useful, but where it is, it can be implemented rather trivially with `ctx.persistentChatAction`:

```TS
// sendChatAction will be called for 10 seconds
await ctx.persistentChatAction("typing", () => sleep(10000));
```